File: samples/client/petstore/cpp-tiny/lib/Models/Node.cpp


#include "Node.h"

using namespace Tiny;

Node::Node()
{
	left = Node();
	right = Node();
}

Node::Node(std::string jsonString)
{
	this->fromJson(jsonString);
}

Node::~Node()
{

}

void
Node::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *leftKey = "left";

    if(object.has_key(leftKey))
    {
        bourne::json value = object[leftKey];




        Node* obj = &left;
		obj->fromJson(value.dump());

    }

    const char *rightKey = "right";

    if(object.has_key(rightKey))
    {
        bourne::json value = object[rightKey];




        Node* obj = &right;
		obj->fromJson(value.dump());

    }


}

bourne::json
Node::toJson()
{
    bourne::json object = bourne::json::object();






	object["left"] = getLeft().toJson();






	object["right"] = getRight().toJson();


    return object;

}

Node
Node::getLeft()
{
	return left;
}

void
Node::setLeft(Node  left)
{
	this->left = left;
}

Node
Node::getRight()
{
	return right;
}

void
Node::setRight(Node  right)
{
	this->right = right;
}

File: samples/server/petstore/java-helidon-server/se/src/main/java/org/openapitools/server/model/Node.java
package org.openapitools.server.model;



/**
 * dummy
 */
public class Node   {

    private Node left;
    private Node right;

    /**
     * Default constructor.
     */
    public Node() {
    // JSON-B / Jackson
    }

    /**
     * Create Node.
     *
     * @param left left
     * @param right right
     */
    public Node(
        Node left, 
        Node right
    ) {
        this.left = left;
        this.right = right;
    }



    /**
     * Get left
     * @return left
     */
    public Node getLeft() {
        return left;
    }

    public void setLeft(Node left) {
        this.left = left;
    }

    /**
     * Get right
     * @return right
     */
    public Node getRight() {
        return right;
    }

    public void setRight(Node right) {
        this.right = right;
    }

    /**
      * Create a string representation of this pojo.
    **/
    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("class Node {\n");
        
        sb.append("    left: ").append(toIndentedString(left)).append("\n");
        sb.append("    right: ").append(toIndentedString(right)).append("\n");
        sb.append("}");
        return sb.toString();
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces
     * (except the first line).
    */
    private static String toIndentedString(Object o) {
        if (o == null) {
          return "null";
        }
        return o.toString().replace("\n", "\n    ");
    }
}

